From 82176ad1b811e92cc8ae8d8f7fc1d0a8b2e6325a Mon Sep 17 00:00:00 2001
From: Benoit Bovy <benbovy@gmail.com>
Date: Tue, 20 Aug 2024 14:56:33 +0200
Subject: [PATCH] add openssl as requirement

Not only for special cases `BUNDLED` and `BREW` for `S2_SOURCE`. Using
`S2_SOURCE=SYSTEM`, there may be cases where openssl and s2geometry were
installed in different prefix paths (e.g., system-installed openssl and
custom s2geometry installation). The change here should address those
cases too.
---
 CMakeLists.txt | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f71ff49..518d124 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,11 +114,6 @@ macro(build_s2)

   set_property(TARGET s2 PROPERTY CXX_STANDARD ${CMAKE_CXX_STANDARD})

-  # this might be needed since s2geometry includes it in general
-  # but not for any target explicilty?
-  find_package(OpenSSL)
-  target_include_directories(s2 INTERFACE ${OPENSSL_INCLUDE_DIR})
-
   get_target_property(S2_VERSION_STRING s2 VERSION)
   extract_s2_version(${S2_VERSION_STRING})
   add_library(s2::s2 ALIAS s2)
@@ -128,9 +123,6 @@ if(${S2GEOGRAPHY_S2_SOURCE} STREQUAL "CONDA")
   set(S2_ROOT_DIR "$ENV{CONDA_PREFIX}")
   set(S2_SOURCE "SYSTEM")
 elseif(${S2GEOGRAPHY_S2_SOURCE} STREQUAL "BREW")
-  # required for Homebrew installed s2geometry headers to find OpenSSL headers
-  find_package(OpenSSL)
-  include_directories(${OPENSSL_INCLUDE_DIR})
   set(S2_SOURCE "SYSTEM")
 else()
   set(S2_SOURCE ${S2GEOGRAPHY_S2_SOURCE})
@@ -159,4 +151,9 @@ if (MSVC AND NOT ${S2_SOURCE} STREQUAL "BUNDLED")
   target_compile_options(s2::s2 INTERFACE /J)
 endif()

+# this might be needed since s2geometry includes it in general
+# but not for any target explicilty?
+find_package(OpenSSL REQUIRED)
+target_include_directories(s2::s2 INTERFACE ${OPENSSL_INCLUDE_DIR})
+
 # --- Abseil (bundled build not supported)

 find_package(absl REQUIRED)
--
2.36.0
